mathesar/api/serializers/shared_serializers.py

```diff
@@ -115,118 +112,23 @@ class NumberDisplayOptionSerializer(MathesarErrorMessageMixin, OverrideRootParti
-
-
-class TimeWithTimeZoneFormatValidator(AbstractDateTimeFormatValidator):
+def raise_if_duration_format_invalid(format):
+    if 'z' in format.lower():
+        raise serializers.ValidationError(
+            "Duration column cannot contain timezone display format"
+        )
 
-    def validate(self, datetime_obj, display_format, serializer_field):
-        time_only_format = 'HH:mm:ss.SSSZZ'
-        time_str = arrow.get('2013-09-30T15:34:00.000-07:00').format(time_only_format)
-        parsed_time_str = arrow.get(time_str, time_only_format)
-        if parsed_time_str.date() != datetime_obj.date():
-            raise serializers.ValidationError("Time column cannot contain date display format")
 
-
-class TimeWithoutTimeZoneFormatValidator(TimeWithTimeZoneFormatValidator):
-
-    def validate(self, datetime_obj, display_format, serializer_field):
-        if 'z' in display_format.lower():
-            raise serializers.ValidationError("Time without timezone column cannot contain timezone display format")
-        return super().validate(datetime_obj, display_format, serializer_field)
-
-
-class DurationFormatValidator(AbstractDateTimeFormatValidator):
-
-    def validate(self, datetime_obj, display_format, serializer_field):
-        if 'z' in display_format.lower():
-            raise serializers.ValidationError(
-                "Duration column cannot contain timezone display format"
-            )
-
-
-class DateDisplayOptionSerializer(MathesarErrorMessageMixin, OverrideRootPartialMixin, serializers.Serializer):
-    format = serializers.CharField(validators=[DateFormatValidator()])
-
-
-class TimestampWithoutTimezoneDisplayOptionSerializer(
-    MathesarErrorMessageMixin,
-    OverrideRootPartialMixin,
-    serializers.Serializer
-):
-    format = serializers.CharField(validators=[TimestampWithoutTimeZoneFormatValidator()])
-
-
-class TimestampWithTimezoneDisplayOptionSerializer(
+class TimeFormatDisplayOptionSerializer(
     MathesarErrorMessageMixin,
     OverrideRootPartialMixin,
     serializers.Serializer
 ):
-    format = serializers.CharField(validators=[TimestampWithTimeZoneFormatValidator()])
-
-
-class TimeWithTimezoneDisplayOptionSerializer(
-    MathesarErrorMessageMixin,
-    OverrideRootPartialMixin,
-    serializers.Serializer
-):
-    format = serializers.CharField(validators=[TimeWithTimeZoneFormatValidator()])
-
-
-class TimeWithoutTimezoneDisplayOptionSerializer(
-    MathesarErrorMessageMixin,
-    OverrideRootPartialMixin,
-    serializers.Serializer
-):
-    format = serializers.CharField(validators=[TimeWithoutTimeZoneFormatValidator()])
+    format = serializers.CharField(max_length=255)
 
 
 class DurationDisplayOptionSerializer(MathesarErrorMessageMixin, OverrideRootPartialMixin, serializers.Serializer):
-    format = serializers.CharField(validators=[DurationFormatValidator()])
+    format = serializers.CharField(max_length=255, validators=[raise_if_duration_format_invalid])
```

Comment:
Based on our [conclusion](https://wiki.mathesar.org/en/meeting-notes/2022-03#h-2022-03-01), other than a character length check, we don't need any additional validators. So the `validators` field  and the validation function should be removed